keymaster/ta/attestation.c

```diff
@@ -162,7 +163,231 @@ TEE_Result TA_open_root_ec_attest_cert(TEE_ObjectHandle *attCert)
 	return res;
 }
 
-TEE_Result TA_create_rsa_attest_key(void)
+#ifdef CFG_ATTESTATION_PROVISIONING
+static TEE_Result TA_set_rsa_attest_key(TEE_TASessionHandle sessionSTA, keymaster_blob_t key_data)
+{
+	TEE_Result result = TEE_ERROR_BAD_PARAMETERS;
+	TEE_ObjectHandle RSAobject = TEE_HANDLE_NULL;
+	TEE_Attribute	 	*attrs;
+	uint32_t			attrs_count;
+	uint64_t key_rsa_public_exponent = UNDEFINED;
+	uint32_t key_size = UNDEFINED;
+
+	DMSG("RSA root attestation key ...");
+
+	if (TA_decode_pkcs8(sessionSTA, key_data, &attrs,
+			&attrs_count, KM_ALGORITHM_RSA, &key_size,
+			&key_rsa_public_exponent) != KM_ERROR_OK)
+	{
+		goto error_1;
+	}
+
+	if (key_size % 8 != 0 || key_size > MAX_KEY_RSA) {
+		EMSG("RSA key size %d must be multiple of 8 and less than %u",
+							key_size,MAX_KEY_RSA);
+		goto error_2;
+	}
+
+	//Create object in storage
+	/* TODO : store in RPMB */
```

Comment:
btw, just replacing `TEE_STORAGE_PRIVATE` with `TEE_STORAGE_PRIVATE_RPMB` in the line below doesn't work?